File: ABC/ABC059/C.py
n = int(input())
a = list(map(int, input().split()))


def sign(a: int) -> int:
    return (a > 0) - (a < 0)


# +
ans1 = 0 if a[0] > 0 else 1 - a[0]
prs1 = a[0] if a[0] > 0 else 1
for i in a[1:]:
    if sign(-prs1) == sign(prs1 + i):
        prs1 += i
    else:
        trg1 = -prs1 + sign(-prs1)
        dif1 = max(trg1, i) - min(trg1, i)
        ans1 += dif1
        prs1 = sign(-prs1)

# -
ans2 = 0 if a[0] < 0 else a[0] + 1
prs2 = a[0] if a[0] < 0 else -1
for i in a[1:]:
    if sign(-prs2) == sign(prs2 + i):
        prs2 += i
    else:
        trg2 = -prs2 + sign(-prs2)
        dif2 = max(trg2, i) - min(trg2, i)
        ans2 += dif2
        prs2 = sign(-prs2)

print(min(ans1, ans2))

File: ABC/ABC059/C.cpp
#include <bits/stdc++.h>
#include <atcoder/all>
using namespace std;
using namespace atcoder;

#define fs(n) fixed << setprecision(n)
#define mp(a, b) make_pair(a, b)
#define all(x) x.begin(), x.end()
#define const constexpr
#define pdesc(t) t, vector<t>, greater<t>
using ll = long long;
using ld = long double;

ll modv = 998244353;

int main(){
  int n; cin >> n;
  vector<ll> a(n);
  for(int i = 0; i < n; i++) cin >> a[i];

  auto sign = [](ll a)->ll{return (a > 0) - (a < 0);};

  // +
  ll ans1 = 0;
  ll psc = a[0];
  for(int i = 1; i < n; i++){
    if(abs(a[i]) > abs(psc) && sign(a[i]) != sign(psc)){
      psc += a[i];
    }else{
      ll trg = -psc + sign(-psc);
      ll dif = max(a[i], trg) - min(a[i], trg);
      ans1 += dif;
      psc += trg;
    }
  }

  // -
  ll ans2 = 0;
  ll psc = a[0];
  for(int i = 1; i < n; i++){
    if(abs(a[i]) > abs(psc) && sign(a[i]) != sign(psc)){
      psc += a[i];
    }else{
      ll trg = -psc + sign(-psc);
      ll dif = max(a[i], trg) - min(a[i], trg);
      ans2 += dif;
      psc += trg;
    }
  }
  
  cout << ans2 << endl;
}